spec/lib/services/pollers-api-service-spec.js

```diff
@@ -0,0 +1,251 @@
+// Copyright 2016, EMC, Inc.
+/* jshint node:true */
+
+"use strict";
+
+describe("Http.Services.Api.Pollers", function () {
+    var pollerService;
+    var waterline;
+    var taskProtocol;
+
+    before("Http.Services.Api.Pollers before", function() {
+        helper.setupInjector([
+            helper.require("/lib/services/pollers-api-service.js")
+        ]);
+
+        waterline = helper.injector.get("Services.Waterline");
+        waterline.workitems = {
+            needByIdentifier: sinon.stub().resolves(),
+            find: sinon.stub().resolves([]),
+            create: sinon.stub().resolves(),
+            updateByIdentifier: sinon.stub().resolves(),
+            destroyByIdentifier: sinon.stub().resolves()
+        };
+
+	taskProtocol = helper.injector.get("Protocol.Task");
+        taskProtocol.requestPollerCache = sinon.stub();
+        pollerService = helper.injector.get("Http.Services.Api.Pollers");
+
+    });
+
+    describe("getPollers", function() {
+        it("should expose the appropriate methods", function() {
+            pollerService.should.have.property("getPollers")
+                .that.is.a("function").with.length(1);
+        });
+
+        it("Run getPollers", function() {
+            var mockPoller = [{
+                id: "4532",
+                name: "Pollers.IPMI",
+                config: {}
+            }];
+
+            waterline.workitems.find.resolves(mockPoller);
+            return pollerService.getPollers().then(function (pollers) {
+                expect(pollers).to.deep.equal(mockPoller);
+            });
+        });
+        it('should return error if poller informations is not found', function () {
+            var mockPoller = {"message":"NotFoundError: Could not find workitem with identifier"};
```

Comment:
@srinia6 I think we want to make this an error object: `var mockPollerError = new Errors.NotFoundError('Could not find workitem with identifier')`.